src/routines/mpilib20_reduce.F90

```diff
@@ -0,0 +1,83 @@
+!> Wrapping of the MPI_REDUCE methods
+!> TODOs
+!> Unravel how MPI_Datatype operates
+!> Overload for MPI_IN_PLACE?
```

Comment:
I'm not entirely sure how `MPI_IN_PLACE` works. For a single communicator, it's a pointless operation. Once they're split, we'd need to consider its use

Reply:
MPI_IN_PLACE is actually quite complicated to overload - `MPI_IN_PLACE` replaces sendbuf for the root process (indicating that the sendbuf and recvbuf are the same) BUT on non-root processes, there needs to be a variable passed via sendbuf. Not sure what this means for overloading for when sendbuf is MPI_IN_PLACE, probably would have MPI_IN_PLACE as an optional logical with logic to ensure correct filling of the (in this case) sendbuf and recvbuf variables for MPI_REDUCE.

https://stackoverflow.com/questions/17741574/in-place-mpi-reduce-crashes-with-openmpi

Reply:
Ok, makes sense to ignore in the first instance and add an issue to address. I've never needed to use personally and only seen it recently

- [ ] Create issue